File: frontend/src/types/auditLog.ts
import { PageResponse } from "./api";

/** Backend AuditLogDto sözleşmesi */
export interface AuditLog {
  id: number;
  createdAt: string;
  actorUserId: number | null;
  actorName: string | null;
  actorEmail: string | null;
  actorRole: string | null;
  action: string;
  entityType: string | null;
  entityId: number | null;
  entityDisplay: string | null;
  description: string;
  source: string;
  ipAddress: string | null;
  success: boolean;
  metadata: Record<string, unknown> | null;
}

export type AuditLogPage = PageResponse<AuditLog>;

export type AuditSource = "WEB" | "WHATSAPP" | "SYSTEM";

export interface AuditLogFilters {
  page?: number;
  size?: number;
  action?: string;
  entityType?: string;
  actorUserId?: number;
  source?: string;
  success?: boolean;
  dateFrom?: string;
  dateTo?: string;
  search?: string;
}

File: backend/src/main/java/com/servis/backend/entity/AuditLog.java
package com.servis.backend.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "audit_logs")
public class AuditLog {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false, nullable = false)
    private LocalDateTime createdAt;

    @Column(name = "actor_user_id")
    private Long actorUserId;

    @Column(name = "actor_name", length = 150)
    private String actorName;

    @Column(name = "actor_email", length = 150)
    private String actorEmail;

    @Column(name = "actor_role", length = 50)
    private String actorRole;

    @Column(nullable = false, length = 80)
    private String action;

    @Column(name = "entity_type", length = 80)
    private String entityType;

    @Column(name = "entity_id")
    private Long entityId;

    @Column(name = "entity_display", length = 255)
    private String entityDisplay;

    @Column(nullable = false, length = 500)
    private String description;

    @Column(nullable = false, length = 30)
    private String source = "SYSTEM";

    @Column(name = "ip_address", length = 64)
    private String ipAddress;

    @Column(nullable = false)
    private Boolean success = true;

    @Column(name = "metadata_json", columnDefinition = "TEXT")
    private String metadataJson;
}
